homeassistant/components/fritzbox/climate.py

```diff
@@ -113,18 +113,18 @@ def hvac_mode(self) -> str:
             OFF_REPORT_SET_TEMPERATURE,
             OFF_API_TEMPERATURE,
         ):
-            return HVAC_MODE_OFF
+            return HVACMode.OFF
 
-        return HVAC_MODE_HEAT
+        return HVACMode.HEAT
 
     @property
-    def hvac_modes(self) -> list[str]:
+    def hvac_modes(self) -> list[HVACMode]:  # type:ignore[override]
```

Comment:
Why is this needed?
Is it because the base component sets the type to `list[HVACMode | str]`?
How about changing base component to `list[HVACMode] | list[str]` instead, then it should be fine to override with `list[HVACMode]`?

Reply:
```suggestion
    def hvac_modes(self) -> list[HVACMode]:
```